posts/2023-08-23-liberty-olo-rh-sso.adoc

```diff
@@ -0,0 +1,394 @@
+
+Next we need to create users (non-admin) for this realm. We can use them to test social login when RH-SSO is used as OIDC provider, the non-admin users can be used to log in to the client application. 
+
+Following the steps to create a user (with it's password)
+testuser1/testpasswd1
+
+Create a user called "testuser1"
+Here are the steps to create the users:
+
+* After logging in Select Manage -> Users and select Add user.
+* Put in the value "testuser1" for Username, and click on Save.
+* After saving, on the next page, select Credentials tab. Enter the password on the next page and ensure to change 'ON' to 'Off' for 'Temporary'. Click on "Reset Password", and on the confirmation dialog, click on "Change Password".
+* Then go to the "Role Mappings" tab. On the Role Mappings page, under Client Roles' drop-down, select realm-management.
+* After 'realm-management' role is selected, additional boxes such as 'Available Roles' appears. Under Available Roles, search for 'view-realm' and select Add selected.
+* After the role is selected, it appears under 'Assigned Roles' and 'Effective Roles'.
+* Note: Selecting the role is just a basic requirement to allow the user to login to the user's console on RH-SSO. If without any role assigned, the user will get Forbidden error msg on the browser after login.
+
+Use the url below to test the users you just created (testuser1)
+
+https://keycloak-rh-sso.apps.<cluster-name>/auth/admin/Sso-realm/console/
+
+You should be able to log in successfully using the testuser1/testpasswd1.
+
+After logging in, in the General Section, you should see the endpoints. 
+
+Click on the link for the OpenID Endpoint Configuration - and that should point you to -
+https://keycloak-rh-sso.apps.<cluster-name>/auth/realms/sso-realm/.well-known/openid-configuration
+
+This will be needed for the client registration as the discoveryEndpoint later on.
+
+== Create the olapp-sso secret 
+
+Create a new project - called `gsm-test`, and create a secret in that namespace - - (Workloads->Secrets->Create Secret), called `guide-social-media-login-olapp-sso`, using key `oidc-clientId` and value `gsmapp`.
+
+The key name should be in this format <app-name>-olapp-sso. Use the same <app-name> as what you will use while deploying the applicaton using OLO.
+
+For e.g. in this setup, the <app-name> from the yaml file is `guide-social-media-login`, so the secret name is `guide-social-media-login-olapp-sso`
+
+== Deploy the application to Open Shift using the Open Liberty Operator
+
+Since you have already installed the Open Liberty Operator, use the yaml file given below to deploy the Open Liberty App (guide-social-media-login) - for which you created the image using the Dockerfile mentioned above.
+
+Note the name of the application deployed is `guide-social-media-login`, the same name that was used when creating the secret above.
+
+Point to your image location in this section
```

Comment:
Not sure where these commands are meant to be being used? In terminal? In Dockerfile? Is it clear to others?